ui/src/views/authentication/unauthenticated/login/index.js

```diff
@@ -20,9 +22,20 @@ class Login extends Component {
 		evt.preventDefault();
 		const { email, password } = this.state;
 		const { authenticate } = this.props;
-		authenticate.call({ username: email, password });
+		this.setState({
+			error: "",
+		});
+		authenticate
+			.call({ username: email, password })
+			.catch(error => this.handleError(error));
```

Comment:
my 'connector' higher order component is actually designed specifically to store the state of the promise in a consistent way. in the render function we can check if authenticate.rejected is true and if so the error reason can be accessed using authenticate.reason.

Reply:
don't really get it, but I'll check it out and try to fix.

Reply:
It's not well documented but I've used it throughout the app and there's an example here https://github.com/cadc/glue-stack/blob/master/ui/src/api/connector.js#L120

Reply:
yea yea, it's just the errors are quite specific to components - e.g. email error

Reply:
Sorry man, I'm sure there's a good concept there, I just don't know how to reconcile the generic error handling with the specific error cases without making it less explicit.

One thing I don't like about my error handler is this line:
`error.response.data.errors[0].code`
as it couples the view to the api structure. Maybe this should be handled by the connector, and a standardised error code passed back to the View?

Though it does seem like this could be achieved with the authenticator instead, as it makes sense for the authenticator to be coupled to the API structure.

Anyway you decide what you think is best, it's your project man!